GroceryList/app/src/main/java/com/mobileproto/david/grocerylist/EditItemBuilder.java

```diff
@@ -0,0 +1,74 @@
+package com.mobileproto.david.grocerylist;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.text.InputType;
+import android.widget.EditText;
+
+/**
+ * This class is a AlertDialog.Builder that builds a dialog for the user to edit an existing item
+ * in the grocery list. There is a text field to change its text, a delete button, a cancel button,
+ * and a confirm button.
+ */
+public class EditItemBuilder extends AlertDialog.Builder
+{
+
+    private MainActivity parentActivity;
+    private EditText input;
+    private int position;
+
+    // Define the listeners for each of the three buttons
+
+    // If the user clicks confirm, change the value in the ArrayList and update the ArrayAdapter
+    DialogInterface.OnClickListener positiveListener = new DialogInterface.OnClickListener()
+    {
+        @Override
+        public void onClick(DialogInterface dialog, int which)
+        {
+            String text = input.getText().toString();
+            parentActivity.getGroceryItems().set(position, text);
```

Comment:
Hm, not sure this works as expected. Seems like it'd be something run statically when the class is loaded as opposed to when each AlertDialog is built... I could be mistaken. Might be worth checking out if you haven't already.